### PR TITLE
The referrer for a statistics view event is almost always wrong

### LINKED ISSUE #8817: The referrer for a statistics view event is almost always wrong
**Describe the bug**
The referrer tracked in statistics view events is almost always wrong, because [the one from the header of the rest request is used](https://github.com/DSpace/DSpace/blob/d7749e243801364417dcc3618925784a44d877c1/dspace-api/src/main/java/org/dspace/statistics/SolrLoggerServiceImpl.java#L327-L329) rather then the referrer from the client request to get to the page

**To Reproduce**
1. Go to a DSpace item page from a link somewhere else (e.g. a search results page, or a link in a 3rd party site)
2. Either wait for the solr stats core to autocommit, or send a commit request
3. Find the event in the solr stats core (perhaps based on your ip or the time field)
4. Check the referrer field for the solr doc and notice that it's the item page itself, not the page you came from

**Expected behavior**
The UI should have the option to send the referrer along with [the view event](https://github.com/DSpace/RestContract/blob/main/statistics-viewevents.md#view-events-endpoint), and if it's present the rest api should use it.

For other applications I can imagine it can still be useful to use the actual referrer from the rest request, so I'd still leave that in as a fallback for when the new referrer param is missing

### PR DESCRIPTION
**Pull request by benbosman**
## References
* Fixes #8817
* Also requires https://github.com/DSpace/dspace-angular/pull/2288

## Description
This adds the support in REST to define a Referrer when creating a view event

## Instructions for Reviewers
This can be tested when clicking from an external site to your repository, or between pages in your repository
Keep in mind that when clicking from an external site, your browser will most likely limit the referrer to the hostname

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_

- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & integration tests). Exceptions may be made if previously agreed upon.
- [x] My PR passes Checkstyle validation based on the [Code Style Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Style+Guide).
- [x] My PR includes Javadoc for _all new (or modified) public methods and classes_. It also includes Javadoc for large or complex private methods.
- [x] My PR passes all tests and includes new/updated Unit or Integration Tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new libraries/dependencies (in any `pom.xml`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR modifies REST API endpoints, I've opened a separate [REST Contract](https://github.com/DSpace/RestContract/blob/main/README.md) PR related to this change.
- [x] If my PR includes new configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

File: dspace-api/src/main/java/org/dspace/statistics/SolrLoggerServiceImpl.java [hunk 1]
```diff
@@ -213,14 +213,20 @@ public void post(DSpaceObject dspaceObject, HttpServletRequest request,
...
     public void postView(DSpaceObject dspaceObject, HttpServletRequest request,
                          EPerson currentUser) {
-        if (solr == null) {
+        postView(dspaceObject, request, currentUser, null);
+    }
+
+    @Override
+    public void postView(DSpaceObject dspaceObject, HttpServletRequest request,
+                         EPerson currentUser, String referrer) {
+        if (solr == null || locationService == null) {
```
Comment (nwoodward) [inline dspace-api/src/main/java/org/dspace/statistics/SolrLoggerServiceImpl.java:222]: locationService is no longer required after https://github.com/DSpace/DSpace/pull/8571.

**Review by tdonohue — CHANGES_REQUESTED** [review]
@benbosman :  This looks good and works overall, except that it accidentally re-introduces the bug described in #7822.  The fix is simple and is described inline below. You just need to remove all the `locationService == null` checks that you've accidentally added back in.

Also, we need a small PR to the REST contract for this. 

I've verified locally that, if I remove the `locationService` checks, then this works properly.  So, it looks good besides that.

**Review by tdonohue — APPROVED** [review]
👍 Thanks @YanaDePauw and @Atmire-Kristof and @benbosman .  My feedback has been addressed, and I've verified this works with the frontend and ensures the referrer is now correct in `statistics_type:view` entries in Solr:
```
{
        "ip":"[redacted]",
        "referrer":"http://localhost:4000/search?query=testing",
        ...
        "id":"16228f28-2f7e-44ec-9aa5-5898f89e8993",
        "type":2,
        "owningColl":["c2aecee3-d59a-4e9c-9659-34283a719cda"],
        "owningComm":["c0e4de93-f506-4990-a840-d406f6f2ada7"],
        "time":"2023-06-16T15:21:32.135Z",
        "statistics_type":"view",
        "uid":"8cd6bd9d-cbfb-4e99-b691-11ad872d1bea"},
```

**Review by atarix83 — APPROVED** [review]
Thanks @benbosman 

i've tested with angular PR and it works properly

**Pull request merged**